### DISCUSSION TITLE
VM stuck at "waiting for capacity"

### DISCUSSION THREAD

**Discussion by rlimberger** [ubicloud/ubicloud#1995] [General]
I cloudyfied a Hetzner server, created a project and VM. However, the VM is stuck in "waiting for capacity". How can I find out what the reason for this is and how can I resolve it? Thank you.

I see this in the logs:
`ubicloud-app          | 01:21:09 respirate.1 | {"lack_of_capacity":{"location":"hetzner-fsn1","arch":"x64","family":"standard","queue_size":1},"message":"No capacity left","time":"2024-09-29 01:21:09 +0000","thread":"vmjvnxxsy8166pkt0vq9mdpamn"}`

**Comment by enescakir** [comment]
You need to download the boot image to your VM host. You can use `pry` for this.

```ruby
# Get your VmHost. If you have multiple ones, you can select by ID etc.
vm_host = VmHost.first

# I assume your vm's boot image is `ubuntu-noble`. You can see available boot images here: https://github.com/ubicloud/ubicloud/blob/cfd27208cb51c3c989e3b9760593cb402b53d1da/prog/download_boot_image.rb#L76
st = vmh.download_boot_image("ubuntu-noble")

# Wait to download. It will be deleted and raise `No Record Found` when done
st.reload

**Comment by rlimberger** [comment]
Im sorry Enes, I am absolutely new to Ubicloud, so please excuse my question. 

After I setup a VM in the Ubicloud dashboard at localhost:3000, I should ssh into the Hetzner host and run pry to get the boot image?

**Comment by furkansahin** [comment]
Looks like we missed the mark here. Still responding just in case it helps someone else. You don't need to ssh into the host and run pry @rlimberger . You just need to run the above commands that Enes shared in your local console. The exact console that you used to add the host. Or, now we have a new parameter that you can use during host creation.
```
Prog::Vm::HostNexus.assemble(..., default_boot_images: ["ubuntu-noble", "ubuntu-jammy"])
```
should do the trick.

**Comment by nnWhisperer** [comment]
I'm having the same problem:
```
ubicloud-app          | 14:24:40 respirate.1 | {"lack_of_capacity":{"location":"hetzner-fsn1","arch":"x64","family":"standard","queue_size":1},"message":"No capacity left","time":"2025-05-15 14:24:33 +0000","thread":"vmkeh849bp8y7gjx05k1h1zjyz"}
```
When I run ./bin/pry I got an error about `not having ruby`. I installed it with apt, which only installed 3.0, in the gemfile it says 3.4
When I run ./bin.pry now, it says:
```
./bin/pry:29: syntax error, unexpected ')'
  UBID.decode(*)
./bin/pry:40: module definition in method body
  module PryReloader
./bin/pry:55: syntax error, unexpected end-of-input, expecting `end'
```
As far as I understood, ubicloud wants users to only use their platform, not this repo.

**Reply by fdr** [reply]
The way we handle our development environments is with `mise`, per https://github.com/ubicloud/ubicloud/blob/main/DEVELOPERS.md, you do need a correct version of ruby. We routinely deploy with the Heroku Build Pack, which handles getting the right version. We don't have any particular procedure for handling that on unassisted debian, perhaps you could use `ruby-build` (or handle the tarball yourself) for that. There are also some PPAs.
